### PR TITLE
Rename RequestResponsePair to HttpExchange

### PR DESCRIPTION
**Pull request by fornwall**
I think `HttpExchange` reads a bit better than `RequestResponsePair`, what do you think?

Regarding naming, what do you think about prefixing all the types with `Http`?

- `Headers` -> `HttpHeaders`
- `Query` -> `HttpQuery`
- `Protocol` -> `HttpProtocol`
- `HttpMethod` (is already prefixed)
- `Request` -> `HttpRequest`
- `Response` -> `HttpResponse`

I also tried out a `from_jsonl()` method - it doesn't do that much work here in python, but I think we might want to have a `from_jsonl()`/`to_jsonl()` pair of functions generally in every language implementation of "HTTP Types", to encourage working with the JSON format.

### CODE REVIEW

File: http_types/utils.py [hunk 1]
```diff
@@ -205,15 +205,25 @@ def from_dict(obj: Any) -> RequestResponsePair:
...
 
         Arguments:
-            reqres {RequestResponsePair} -- Possible request-response object.
+            reqres {HttpExchange} -- Possible request-response object.
         """
-        check_type("request-response", reqres, RequestResponsePair)
+        check_type("request-response", reqres, HttpExchange)
+
+    @staticmethod
+    def from_jsonl(input_file: IO) -> Generator[HttpExchange, None, None]:
```
Comment (ksaaskil) [inline http_types/utils.py:222]: `Generator[HttpExchange, None, None]` could be replaced by `Iterator[HttpExchange]`?

Comment (ksaaskil) [reply http_types/utils.py:222]: Also, should the input be `TextIO` or does it also work for binary I/O?

**Review by ksaaskil — APPROVED** [review]
I'm fine with this! Note that the change will break code in `meeshkan` and some internal research code that @MariaMalitckaya @nakulbansal1609 are working on so pinging them as well. Regarding the proposed names, I'm fine with those as well.

File: http_types/utils.py [hunk 2]
```diff
@@ -205,15 +205,25 @@ def from_dict(obj: Any) -> RequestResponsePair:
...
+        """Read Http exchanges line by line from a file-like object.
+
+        Arguments:
+            input_file: {IO} -- The input to read from.
+        """
+        for line in input_file:
+            yield HttpExchangeBuilder.from_dict(json.loads(line))
```
Comment (ksaaskil) [inline http_types/utils.py:229]: Is this the same as `return (HttpExchangeBuilder.from_dict(json.loads(line)) for line in input_file)`?  Your version is more readable and therefore more pythonic, just wondering.

File: http_types/utils.py (hunk 1 above)
Comment (MariaMalitckaya) [reply http_types/utils.py:222]: It is fine for me now! It is quite easy to change it. But I think @nakulbansal1609  and @aby2s  will have more work to do.

File: http_types/utils.py (hunk 2 above)
Comment (fornwall) [reply http_types/utils.py:229]: It seems to be the same from what I can tell with experimenting - both creates a type that is an `isinstance(ret, types.GeneratorType)` and generates the same values.

**Pull request merged**